src/server/scripts/Spells/spell_mage.cpp

```diff
@@ -695,6 +701,84 @@ class spell_mage_water_elemental_freeze : public SpellScript
     }
 };
 
+// Ice Lance - 30455
+class spell_mage_ice_lance : public SpellScript
+{
+    PrepareSpellScript(spell_mage_ice_lance);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({
+            SPELL_MAGE_ICE_LANCE,
+            SPELL_MAGE_ICE_LANCE_TRIGGER
+        });
+    }
+
+    void HandleOnHit(SpellEffIndex /*effIndex*/)
+    {
+        Unit* caster = GetCaster();
+        Unit* target = GetHitUnit();
+        if (!caster && !target)
```

Comment:
In `SpellScript` "Hit" hook caster is never null (target might be if you are hitting a gameobject or item)